pkg/apis/core/validation/validation.go

```diff
@@ -6117,6 +6107,41 @@ func ValidatePodResize(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		allErrs = append(allErrs, field.Forbidden(specPath, "Pod running on node without support for resize"))
 	}
 
+	// Part 2: Validate that changes between pod-level resources in oldPod.Spec.Resources and
+	// newPod.Spec.Resources are allowed.
+
+	isPodLevelResourcesSet := func(pod *core.Pod) bool {
+		return pod.Spec.Resources != nil &&
+			(len(pod.Spec.Resources.Requests)+len(pod.Spec.Resources.Limits) > 0)
+	}
+
+	originalCPUMemPodSpec := *newPod.Spec.DeepCopy()
+
+	if isPodLevelResourcesSet(oldPod) || isPodLevelResourcesSet(newPod) {
+		// pods with pod-level resources cannot be resized without
+		// InPlacePodLevelResourcesVerticalScaling feature gate being enabled.
+		if !opts.InPlacePodLevelResourcesVerticalScalingEnabled {
+			return field.ErrorList{field.Forbidden(field.NewPath(""), "pods with pod-level resources is forbidden when InPlacePodLevelResourcesVerticalScalingEnabled feature gate is disabled")}
+		}
+
+		if !opts.PodLevelResourcesEnabled {
+			return field.ErrorList{field.Forbidden(field.NewPath(""), "pods with pod-level resources is forbidden when PodLevelResources feature gate is disabled")}
+		}
```

Comment:
I think this will no longer be needed after you declare the feature gate dependency